// @chainlink/contracts/ethers/v0.6/factories/GasGuzzler__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GasGuzzler__factory = void 0;
const ethers_1 = require("ethers");
class GasGuzzler__factory extends ethers_1.ContractFactory {
    constructor(signer) {
        super(_abi, _bytecode, signer);
    }
    deploy(overrides) {
        return super.deploy(overrides);
    }
    getDeployTransaction(overrides) {
        return super.getDeployTransaction(overrides);
    }
    attach(address) {
        return super.attach(address);
    }
    connect(signer) {
        return super.connect(signer);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.GasGuzzler__factory = GasGuzzler__factory;
const _abi = [
    {
        stateMutability: "payable",
        type: "fallback",
    },
];
const _bytecode = "0x6080604052348015600f57600080fd5b50601680601d6000396000f3fe60806040525b600556fea164736f6c6343000606000a";
