// @chainlink/contracts/ethers/v0.6/factories/LinkTokenReceiver__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LinkTokenReceiver__factory = void 0;
const ethers_1 = require("ethers");
class LinkTokenReceiver__factory {
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.LinkTokenReceiver__factory = LinkTokenReceiver__factory;
const _abi = [
    {
        inputs: [],
        name: "getChainlinkToken",
        outputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "_sender",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "_amount",
                type: "uint256",
            },
            {
                internalType: "bytes",
                name: "_data",
                type: "bytes",
            },
        ],
        name: "onTokenTransfer",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
];
