// @chainlink/contracts/ethers/v0.8/factories/KeeperCompatibleInterface__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Signer } from "ethers";
import { Provider } from "ethers/providers";

import { KeeperCompatibleInterface } from "../KeeperCompatibleInterface";

export class KeeperCompatibleInterface__factory {
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): KeeperCompatibleInterface {
    return new Contract(
      address,
      _abi,
      signerOrProvider
    ) as KeeperCompatibleInterface;
  }
}

const _abi = [
  {
    inputs: [
      {
        internalType: "bytes",
        name: "checkData",
        type: "bytes",
      },
    ],
    name: "checkUpkeep",
    outputs: [
      {
        internalType: "bool",
        name: "upkeepNeeded",
        type: "bool",
      },
      {
        internalType: "bytes",
        name: "performData",
        type: "bytes",
      },
    ],
    stateMutability: "nonpayable",
    type: "function",
  },
  {
    inputs: [
      {
        internalType: "bytes",
        name: "performData",
        type: "bytes",
      },
    ],
    name: "performUpkeep",
    outputs: [],
    stateMutability: "nonpayable",
    type: "function",
  },
];
