// @chainlink/contracts/ethers/v0.5/factories/MaliciousChainlinkClient__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, ContractFactory, Signer } from "ethers";
import { Provider } from "ethers/providers";
import { UnsignedTransaction } from "ethers/utils/transaction";

import { TransactionOverrides } from "..";
import { MaliciousChainlinkClient } from "../MaliciousChainlinkClient";

export class MaliciousChainlinkClient__factory extends ContractFactory {
  constructor(signer?: Signer) {
    super(_abi, _bytecode, signer);
  }

  deploy(overrides?: TransactionOverrides): Promise<MaliciousChainlinkClient> {
    return super.deploy(overrides) as Promise<MaliciousChainlinkClient>;
  }
  getDeployTransaction(overrides?: TransactionOverrides): UnsignedTransaction {
    return super.getDeployTransaction(overrides);
  }
  attach(address: string): MaliciousChainlinkClient {
    return super.attach(address) as MaliciousChainlinkClient;
  }
  connect(signer: Signer): MaliciousChainlinkClient__factory {
    return super.connect(signer) as MaliciousChainlinkClient__factory;
  }
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): MaliciousChainlinkClient {
    return new Contract(
      address,
      _abi,
      signerOrProvider
    ) as MaliciousChainlinkClient;
  }
}

const _abi = [
  {
    anonymous: false,
    inputs: [
      {
        indexed: true,
        name: "id",
        type: "bytes32",
      },
    ],
    name: "ChainlinkRequested",
    type: "event",
  },
  {
    anonymous: false,
    inputs: [
      {
        indexed: true,
        name: "id",
        type: "bytes32",
      },
    ],
    name: "ChainlinkFulfilled",
    type: "event",
  },
  {
    anonymous: false,
    inputs: [
      {
        indexed: true,
        name: "id",
        type: "bytes32",
      },
    ],
    name: "ChainlinkCancelled",
    type: "event",
  },
];

const _bytecode =
  "0x608060405260016004556001600655348015601957600080fd5b5060358060276000396000f3fe6080604052600080fdfea165627a7a72305820c4d4ac205964e7049bcf805631dc9d139b383faa475f23c68a49aa99797e48fa0029";
