// @chainlink/contracts/ethers/v0.4/factories/Chainlinked__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, ContractFactory, Signer } from "ethers";
import { Provider } from "ethers/providers";
import { UnsignedTransaction } from "ethers/utils/transaction";

import { TransactionOverrides } from "..";
import { Chainlinked } from "../Chainlinked";

export class Chainlinked__factory extends ContractFactory {
  constructor(signer?: Signer) {
    super(_abi, _bytecode, signer);
  }

  deploy(overrides?: TransactionOverrides): Promise<Chainlinked> {
    return super.deploy(overrides) as Promise<Chainlinked>;
  }
  getDeployTransaction(overrides?: TransactionOverrides): UnsignedTransaction {
    return super.getDeployTransaction(overrides);
  }
  attach(address: string): Chainlinked {
    return super.attach(address) as Chainlinked;
  }
  connect(signer: Signer): Chainlinked__factory {
    return super.connect(signer) as Chainlinked__factory;
  }
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): Chainlinked {
    return new Contract(address, _abi, signerOrProvider) as Chainlinked;
  }
}

const _abi = [
  {
    anonymous: false,
    inputs: [
      {
        indexed: true,
        name: "id",
        type: "bytes32",
      },
    ],
    name: "ChainlinkRequested",
    type: "event",
  },
  {
    anonymous: false,
    inputs: [
      {
        indexed: true,
        name: "id",
        type: "bytes32",
      },
    ],
    name: "ChainlinkFulfilled",
    type: "event",
  },
  {
    anonymous: false,
    inputs: [
      {
        indexed: true,
        name: "id",
        type: "bytes32",
      },
    ],
    name: "ChainlinkCancelled",
    type: "event",
  },
];

const _bytecode =
  "0x60806040526001600455348015601457600080fd5b5060358060226000396000f3006080604052600080fd00a165627a7a7230582030c90ba474cf9c5fe0bcd82b9a567ba57a11caf7c9dd3a190024d68703e7febb0029";
