// @chainlink/contracts/ethers/v0.7/factories/OwnableInterface__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Signer } from "ethers";
import { Provider } from "ethers/providers";

import { OwnableInterface } from "../OwnableInterface";

export class OwnableInterface__factory {
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): OwnableInterface {
    return new Contract(address, _abi, signerOrProvider) as OwnableInterface;
  }
}

const _abi = [
  {
    inputs: [],
    name: "acceptOwnership",
    outputs: [],
    stateMutability: "nonpayable",
    type: "function",
  },
  {
    inputs: [],
    name: "owner",
    outputs: [
      {
        internalType: "address",
        name: "",
        type: "address",
      },
    ],
    stateMutability: "nonpayable",
    type: "function",
  },
  {
    inputs: [
      {
        internalType: "address",
        name: "recipient",
        type: "address",
      },
    ],
    name: "transferOwnership",
    outputs: [],
    stateMutability: "nonpayable",
    type: "function",
  },
];
