// @chainlink/contracts/ethers/v0.7/factories/AuthorizedReceiver__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthorizedReceiver__factory = void 0;
const ethers_1 = require("ethers");
class AuthorizedReceiver__factory {
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.AuthorizedReceiver__factory = AuthorizedReceiver__factory;
const _abi = [
    {
        anonymous: false,
        inputs: [
            {
                indexed: false,
                internalType: "address[]",
                name: "senders",
                type: "address[]",
            },
            {
                indexed: false,
                internalType: "address",
                name: "changedBy",
                type: "address",
            },
        ],
        name: "AuthorizedSendersChanged",
        type: "event",
    },
    {
        inputs: [],
        name: "getAuthorizedSenders",
        outputs: [
            {
                internalType: "address[]",
                name: "",
                type: "address[]",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "sender",
                type: "address",
            },
        ],
        name: "isAuthorizedSender",
        outputs: [
            {
                internalType: "bool",
                name: "",
                type: "bool",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address[]",
                name: "senders",
                type: "address[]",
            },
        ],
        name: "setAuthorizedSenders",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
];
