// @chainlink/contracts/ethers/v0.5/factories/AggregatorV2V3Interface__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AggregatorV2V3Interface__factory = void 0;
const ethers_1 = require("ethers");
class AggregatorV2V3Interface__factory {
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.AggregatorV2V3Interface__factory = AggregatorV2V3Interface__factory;
const _abi = [
    {
        constant: true,
        inputs: [],
        name: "decimals",
        outputs: [
            {
                name: "",
                type: "uint8",
            },
        ],
        payable: false,
        stateMutability: "view",
        type: "function",
    },
    {
        constant: true,
        inputs: [],
        name: "latestAnswer",
        outputs: [
            {
                name: "",
                type: "int256",
            },
        ],
        payable: false,
        stateMutability: "view",
        type: "function",
    },
    {
        constant: true,
        inputs: [],
        name: "version",
        outputs: [
            {
                name: "",
                type: "uint256",
            },
        ],
        payable: false,
        stateMutability: "view",
        type: "function",
    },
    {
        constant: true,
        inputs: [],
        name: "latestRound",
        outputs: [
            {
                name: "",
                type: "uint256",
            },
        ],
        payable: false,
        stateMutability: "view",
        type: "function",
    },
    {
        constant: true,
        inputs: [],
        name: "description",
        outputs: [
            {
                name: "",
                type: "string",
            },
        ],
        payable: false,
        stateMutability: "view",
        type: "function",
    },
    {
        constant: true,
        inputs: [],
        name: "latestTimestamp",
        outputs: [
            {
                name: "",
                type: "uint256",
            },
        ],
        payable: false,
        stateMutability: "view",
        type: "function",
    },
    {
        constant: true,
        inputs: [
            {
                name: "_roundId",
                type: "uint80",
            },
        ],
        name: "getRoundData",
        outputs: [
            {
                name: "roundId",
                type: "uint80",
            },
            {
                name: "answer",
                type: "int256",
            },
            {
                name: "startedAt",
                type: "uint256",
            },
            {
                name: "updatedAt",
                type: "uint256",
            },
            {
                name: "answeredInRound",
                type: "uint80",
            },
        ],
        payable: false,
        stateMutability: "view",
        type: "function",
    },
    {
        constant: true,
        inputs: [
            {
                name: "roundId",
                type: "uint256",
            },
        ],
        name: "getAnswer",
        outputs: [
            {
                name: "",
                type: "int256",
            },
        ],
        payable: false,
        stateMutability: "view",
        type: "function",
    },
    {
        constant: true,
        inputs: [
            {
                name: "roundId",
                type: "uint256",
            },
        ],
        name: "getTimestamp",
        outputs: [
            {
                name: "",
                type: "uint256",
            },
        ],
        payable: false,
        stateMutability: "view",
        type: "function",
    },
    {
        constant: true,
        inputs: [],
        name: "latestRoundData",
        outputs: [
            {
                name: "roundId",
                type: "uint80",
            },
            {
                name: "answer",
                type: "int256",
            },
            {
                name: "startedAt",
                type: "uint256",
            },
            {
                name: "updatedAt",
                type: "uint256",
            },
            {
                name: "answeredInRound",
                type: "uint80",
            },
        ],
        payable: false,
        stateMutability: "view",
        type: "function",
    },
    {
        anonymous: false,
        inputs: [
            {
                indexed: true,
                name: "current",
                type: "int256",
            },
            {
                indexed: true,
                name: "roundId",
                type: "uint256",
            },
            {
                indexed: false,
                name: "timestamp",
                type: "uint256",
            },
        ],
        name: "AnswerUpdated",
        type: "event",
    },
    {
        anonymous: false,
        inputs: [
            {
                indexed: true,
                name: "roundId",
                type: "uint256",
            },
            {
                indexed: true,
                name: "startedBy",
                type: "address",
            },
            {
                indexed: false,
                name: "startedAt",
                type: "uint256",
            },
        ],
        name: "NewRound",
        type: "event",
    },
];
