// @chainlink/contracts/ethers/v0.5/factories/ChainlinkRequestInterface__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChainlinkRequestInterface__factory = void 0;
const ethers_1 = require("ethers");
class ChainlinkRequestInterface__factory {
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.ChainlinkRequestInterface__factory = ChainlinkRequestInterface__factory;
const _abi = [
    {
        constant: false,
        inputs: [
            {
                name: "sender",
                type: "address",
            },
            {
                name: "requestPrice",
                type: "uint256",
            },
            {
                name: "serviceAgreementID",
                type: "bytes32",
            },
            {
                name: "callbackAddress",
                type: "address",
            },
            {
                name: "callbackFunctionId",
                type: "bytes4",
            },
            {
                name: "nonce",
                type: "uint256",
            },
            {
                name: "dataVersion",
                type: "uint256",
            },
            {
                name: "data",
                type: "bytes",
            },
        ],
        name: "oracleRequest",
        outputs: [],
        payable: false,
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        constant: false,
        inputs: [
            {
                name: "requestId",
                type: "bytes32",
            },
            {
                name: "payment",
                type: "uint256",
            },
            {
                name: "callbackFunctionId",
                type: "bytes4",
            },
            {
                name: "expiration",
                type: "uint256",
            },
        ],
        name: "cancelOracleRequest",
        outputs: [],
        payable: false,
        stateMutability: "nonpayable",
        type: "function",
    },
];
