// @chainlink/contracts/ethers/v0.7/factories/AuthorizedReceiverInterface__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthorizedReceiverInterface__factory = void 0;
const ethers_1 = require("ethers");
class AuthorizedReceiverInterface__factory {
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.AuthorizedReceiverInterface__factory = AuthorizedReceiverInterface__factory;
const _abi = [
    {
        inputs: [],
        name: "getAuthorizedSenders",
        outputs: [
            {
                internalType: "address[]",
                name: "",
                type: "address[]",
            },
        ],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "sender",
                type: "address",
            },
        ],
        name: "isAuthorizedSender",
        outputs: [
            {
                internalType: "bool",
                name: "",
                type: "bool",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address[]",
                name: "senders",
                type: "address[]",
            },
        ],
        name: "setAuthorizedSenders",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
];
