// @chainlink/contracts/ethers/v0.7/factories/AuthorizedReceiverInterface__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Signer } from "ethers";
import { Provider } from "ethers/providers";

import { AuthorizedReceiverInterface } from "../AuthorizedReceiverInterface";

export class AuthorizedReceiverInterface__factory {
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): AuthorizedReceiverInterface {
    return new Contract(
      address,
      _abi,
      signerOrProvider
    ) as AuthorizedReceiverInterface;
  }
}

const _abi = [
  {
    inputs: [],
    name: "getAuthorizedSenders",
    outputs: [
      {
        internalType: "address[]",
        name: "",
        type: "address[]",
      },
    ],
    stateMutability: "nonpayable",
    type: "function",
  },
  {
    inputs: [
      {
        internalType: "address",
        name: "sender",
        type: "address",
      },
    ],
    name: "isAuthorizedSender",
    outputs: [
      {
        internalType: "bool",
        name: "",
        type: "bool",
      },
    ],
    stateMutability: "view",
    type: "function",
  },
  {
    inputs: [
      {
        internalType: "address[]",
        name: "senders",
        type: "address[]",
      },
    ],
    name: "setAuthorizedSenders",
    outputs: [],
    stateMutability: "nonpayable",
    type: "function",
  },
];
