// @chainlink/contracts/ethers/v0.4/factories/FlagsInterface__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlagsInterface__factory = void 0;
const ethers_1 = require("ethers");
class FlagsInterface__factory {
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.FlagsInterface__factory = FlagsInterface__factory;
const _abi = [
    {
        constant: false,
        inputs: [
            {
                name: "calldata",
                type: "address[]",
            },
        ],
        name: "lowerFlags",
        outputs: [],
        payable: false,
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        constant: true,
        inputs: [
            {
                name: "",
                type: "address",
            },
        ],
        name: "getFlag",
        outputs: [
            {
                name: "",
                type: "bool",
            },
        ],
        payable: false,
        stateMutability: "view",
        type: "function",
    },
    {
        constant: false,
        inputs: [
            {
                name: "",
                type: "address",
            },
        ],
        name: "setRaisingAccessController",
        outputs: [],
        payable: false,
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        constant: false,
        inputs: [
            {
                name: "calldata",
                type: "address[]",
            },
        ],
        name: "raiseFlags",
        outputs: [],
        payable: false,
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        constant: true,
        inputs: [
            {
                name: "calldata",
                type: "address[]",
            },
        ],
        name: "getFlags",
        outputs: [
            {
                name: "",
                type: "bool[]",
            },
        ],
        payable: false,
        stateMutability: "view",
        type: "function",
    },
    {
        constant: false,
        inputs: [
            {
                name: "",
                type: "address",
            },
        ],
        name: "raiseFlag",
        outputs: [],
        payable: false,
        stateMutability: "nonpayable",
        type: "function",
    },
];
